packages/client/src/components/Browser/DatasetUploader.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import * as tus from 'tus-js-client';
+import gql from 'graphql-tag';
+import { graphql } from 'react-apollo';
+import { compose } from 'recompose';
+import { Upload, Icon } from 'antd';
+import { errorHandler } from 'utils/errorHandler';
+import { getAccessToken } from 'utils/env';
+
+const { Dragger } = Upload;
+
+const DELETE_FILES = gql`
+  mutation deleteFiles(
+    $where: StoreFileWhereInput!
+    $options: StoreFileDeleteOptionInput
+  ) {
+    deleteFiles(where: $where, options: $options)
+  }
+`;
+
+interface Props {
+  groupName: string;
+  datasetId: string;
+  deleteFiles: ({
+    variables,
+  }: {
+    variables: {
+      where: {
+        groupName: string;
+        phfsPrefix: string;
+      };
+    };
+  }) => Promise<{ data: { deleteFiles: number } }>;
+}
+
+function _DatasetUploader(props: Props) {
+  const [files, setFiles] = React.useState([]);
+  const { groupName, datasetId, deleteFiles } = props;
+  const graphqlEndpoint = window.absGraphqlEndpoint
+    ? window.absGraphqlEndpoint
+    : window.graphqlEndpoint;
+  const endpoint = graphqlEndpoint.replace('/graphql', '/tus');
+  const dirpath = `groups/${groupName}/datasets/${datasetId}/`;
+
+  React.useEffect(() => {
+    setFiles([]);
+  }, [groupName, datasetId]);
+
+  function customRequest({ onSuccess, onProgress, onError, file }) {
+    const upload: tus.Upload = file.uploader;
+    upload.options.onError = err => {
+      console.log('Error: ', err);
+      onError(err, file);
+    };
+    upload.options.onProgress = (bytesUploaded, bytesTotal) => {
+      const percentage = ((bytesUploaded / bytesTotal) * 100).toFixed(2);
+      onProgress({ percent: +percentage }, file);
```

Comment:
Is `+percentage` representation to transform to a number?

```suggestion
      onProgress({ percent: +percentage) }, file);
```

Reply:
yes

Reply:
[toFixed](https://developer.mozilla.org/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) method belongs to `Number`, if `bytesUpload` and `bytesTotal` types are string, you can transform here instead of transform at line 57.

You can confirm types at line 55 and add types annotation. By the way, I suggest using `Number` to transform more clear.

```diff
- const percentage = ((bytesUploaded / bytesTotal) * 100).toFixed(2);
+ const percentage = ((Number(bytesUploaded) / Number(bytesTotal)) * 100).toFixed(2);
```

Reply:
Since toFixed returns a string, I would like to replace `+percentage` with `Number(percentage)`